libs/vkd3d/command.c

```diff
@@ -19534,7 +19712,58 @@ static HRESULT d3d12_command_signature_allocate_stream_memory_for_list(
     return S_OK;
 }
 
-static HRESULT d3d12_command_signature_allocate_preprocess_memory_for_list(
+static HRESULT d3d12_command_signature_allocate_preprocess_memory_for_list_ext(
+        struct d3d12_command_list *list,
+        struct d3d12_command_signature *signature, VkPipeline render_pipeline, bool explicit_preprocess,
+        uint32_t max_command_count,
+        struct vkd3d_scratch_allocation *allocation, VkDeviceSize *size)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &list->device->vk_procs;
+    VkGeneratedCommandsMemoryRequirementsInfoEXT info;
+    VkGeneratedCommandsPipelineInfoEXT pipeline_info;
+    VkMemoryRequirements2 memory_info;
+    uint32_t alignment;
+
+    memory_info.sType = VK_STRUCTURE_TYPE_MEMORY_REQUIREMENTS_2;
+    memory_info.pNext = NULL;
+
+    pipeline_info.sType = VK_STRUCTURE_TYPE_GENERATED_COMMANDS_PIPELINE_INFO_EXT;
+    pipeline_info.pNext = NULL;
+    pipeline_info.pipeline = render_pipeline;
```

Comment:
can we just memset these for consistency too?

Reply:
fixed.